timemachine/cpp/src/neighborlist.cu

```diff
@@ -356,8 +364,19 @@ template <typename RealType> int Neighborlist<RealType>::Y() const {
 
 template <typename RealType> int Neighborlist<RealType>::num_row_blocks() const { return ceil_divide(NR_, TILE_SIZE); }
 
+// max_ixn_count determines the number of tile to atom interaction counts. For each tile that interacts with another
+// it can have TILE_SIZE interactions. Note that d_ixn_count_ is only the number of tile interactions, and differs by a factor of TILE_SIZE
 template <typename RealType> int Neighborlist<RealType>::max_ixn_count() const {
-    return num_column_blocks() * num_row_blocks() * WARP_SIZE;
+    // The maximum number of interactions is that of all tiles interacting with all other tiles. Which
+    // is the upper triangular matrix.
+    // Use the maximum NxN matrix to allocate to allow supporting any combination of atom idxs
+    const int n_blocks = ceil_divide(max_size_, TILE_SIZE);
```

Comment:
Looks good, seems conservative: don't think this would be brittle w.r.t. any input geometry, choice of w_coords, or choice of cutoff.

(In the future, hard-coding any smaller of an allocation in this way could be brittle though. If we want to reduce this further in the future, might need to consider something more dynamic, maybe adding a .resize() method or something, so we could start with a small allocation that is not wasteful for instances of typical particle density at default cutoff, but be able to replace it on the fly with a larger allocation on atypical inputs...)

Reply:
One potential thing we can do to save memory is to constrain the memory usage for the IxnGroup potentials, as that is N^2 while it often doesn't need to be.

Reply:
That's a good point, the max number of atom-atom interactions in an IxnGroup potential, N_rows x N_cols, is loosely upper bounded by N^2, but can often be << N^2.